app/src/ui/discard-changes/discard-changes-dialog.tsx

```diff
@@ -57,19 +62,37 @@ export class DiscardChanges extends React.Component<
   }
 
   private getOkButtonLabel() {
-    if (this.props.discardingAllChanges) {
-      return __DARWIN__ ? 'Discard All Changes' : 'Discard all changes'
+    switch (this.props.discardType) {
+      case DiscardType.AllFiles:
+        return __DARWIN__ ? 'Discard All Changes' : 'Discard all changes'
+      case DiscardType.SomeFiles:
+      case DiscardType.Selection:
+        return __DARWIN__ ? 'Discard Changes' : 'Discard changes'
+      default:
+        return assertNever(
+          this.props.discardType,
+          'Invalid discardType property'
+        )
     }
-    return __DARWIN__ ? 'Discard changes' : 'Discard changes'
   }
 
   private getDialogTitle() {
-    if (this.props.discardingAllChanges) {
-      return __DARWIN__
-        ? 'Confirm Discard All Changes'
-        : 'Confirm Discard all changes'
+    switch (this.props.discardType) {
+      case DiscardType.AllFiles:
+        return __DARWIN__
+          ? 'Confirm Discard All Changes'
+          : 'Confirm Discard all changes'
+      case DiscardType.SomeFiles:
+      case DiscardType.Selection:
+        return __DARWIN__
+          ? 'Confirm Discard changes'
+          : 'Confirm Discard changes'
+      default:
+        return assertNever(
+          this.props.discardType,
+          'Invalid discardType property'
+        )
```

Comment:
We typically include the value that we thought was guaranteed to be exhaustively checked in case it happens to fail at runtime. None ever has thus far but theoretically we could be asserting on a value that originates from the API and if that then fails we want to know what new value they've introduced.

```suggestion
        return assertNever(discardType, `Unknown DiscardType: ${discardType}`)
```